src/Table/SynapseTableQueryBuilder.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\TableBackendUtils\Table;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\SQLServer2012Platform;
+use Keboola\TableBackendUtils\Column\ColumnCollection;
+use Keboola\TableBackendUtils\Column\ColumnInterface;
+use Keboola\TableBackendUtils\QueryBuilderException;
+
+class SynapseTableQueryBuilder implements TableQueryBuilderInterface
+{
+    public const MAX_TABLE_COLUMNS = 1024;
+
+    /** @var SQLServer2012Platform|AbstractPlatform */
+    private $platform;
+
+    public function __construct(Connection $connection)
+    {
+        $this->platform = $connection->getDatabasePlatform();
+    }
+
+    public function getCreateTempTableCommand(
+        string $schemaName,
+        string $tableName,
+        ColumnCollection $columns
+    ): string {
+        $this->assertTemporaryTable($tableName);
+        $this->assertTableColumnsCount($columns);
+
+        $columnsSql = [];
+        foreach ($columns as $column) {
+            $columnsSql[] = sprintf(
+                '%s %s',
+                $this->platform->quoteSingleIdentifier($column->getColumnName()),
+                $column->getColumnDefinition()->getSQLDefinition()
+            );
+        }
+
+        return sprintf(
+            'CREATE TABLE %s.%s (%s) WITH (HEAP, LOCATION = USER_DB)',
+            $this->platform->quoteSingleIdentifier($schemaName),
+            $this->platform->quoteSingleIdentifier($tableName),
+            implode(', ', $columnsSql)
+        );
+    }
+
+    private function assertTemporaryTable(string $tableName): void
+    {
+        if (strpos($tableName, '#') !== 0) {
+            throw new QueryBuilderException(
+                sprintf(
+                    'Staging table must start with "#" table name "%s" supplied.',
+                    $tableName
+                ),
+                QueryBuilderException::STRING_CODE_INVALID_TEMP_TABLE
+            );
+        }
+    }
+
+    private function assertTableColumnsCount(ColumnCollection $columns): void
```

Comment:
Tak jako jsem to psal už u connection.
Tahle validace sem imho nepatří. Pokud je to potřeba tak by to mělo být někde o úroveň výš, jako jsem to udělal v KBC v `Storage_Service_BucketBackend_Synapse`

Kdybys tady chtěl něco takového validovat tak bys správně měl i validovat jestli názvy objektů nepřesahují povolenou velikost. A to mi nedává smysl.

Reply:
No pak tu validaci budeš muset mít v connection a všude jinde kde to použiješ :thinking:

Reply:
Lepší místo by bylo, ale pro check v ColumnCollection https://github.com/keboola/php-table-backend-utils/pull/2/commits/47b7321c89ef15e22f44d3d19e35ad3dc9715f96
A check na délku názvu by mohl byt v `SynapseColumn`

Reply:
No nejsem si tím úplně jistý jestli to takhle je ok, uvidím pak až to budu chtít použít v KBC. Každopadně v ColumnCollection je to 100X lepší než v tom sql generatoru 👍